Makefile

```diff
@@ -44,6 +44,10 @@ format:
 	@echo "--> Running go fmt"
 	@go fmt $(PACKAGES)
 
+check-format:
+	@echo "--> Checking format"
+	@! gofmt -l . 2>&1 | read _
+
 test: deps
```

Comment:
I believe this would make it harder for users which run `make test` periodically but do not auto-gofmt their code during development since they would constantly get an error from `check-format`.

Instead, what do you think about making an `all` target like

```
all: check-format test examples
```

and invoke that from Travis CI? This way, we can offer people to just execute `make all` locally and be reassured it'll work during CI as well.